### PR TITLE
Extend TempLED function

### PR DESCRIPTION
**Pull request by tml89**
Set led pin high when brew or steam set point is within range or low (configurable in UserConfig)

### CODE REVIEW

File: src/rancilio-pid.cpp [hunk 1]
```diff
@@ -1675,22 +1681,87 @@ void debugVerboseOutput() {
...
+        blinkLED();
+        return;
+    }
+
+    //Steam || Brew unready 
+    digitalWrite(LEDPIN, brewReadyLedOFF);
 }
 
+static bool previousMode = false;
+void setHardwareLed(bool mode) {
```
Comment (kjyv) [inline src/rancilio-pid.cpp:1758]: wrong indents (should be 4 spaces)

File: src/rancilio-pid.cpp [hunk 2]
```diff
@@ -1835,6 +1906,26 @@ void setup() {
         VoltageSensorOFF = HIGH;
     }
 
+    // Initialize TempLED
+    if (TEMPLED > 0) {
+
```
Comment (kjyv) [inline src/rancilio-pid.cpp:1911]: unnecessary empty line

File: src/rancilio-pid.cpp [hunk 3]
```diff
@@ -1675,22 +1681,87 @@ void debugVerboseOutput() {
...
+    // Blink led on error
+    if (machinestate == kSensorError)
+    {
+        LEDInterval = 100;
+        blinkLED();
+        return;
+    }
+
+    //Steam || Brew unready
```
Comment (kjyv) [inline src/rancilio-pid.cpp:1753]: "not ready"

File: src/rancilio-pid.cpp [hunk 4]
```diff
@@ -201,6 +201,12 @@ boolean setupDone = false;
 int backflushON = 0;      // 1 = activate backflush
 int flushCycles = 0;      // number of active flush cycles
 int backflushState = 10;  // counter for state machine
+int brewReadyLedON, brewReadyLedOFF; // used for brewReady LED
+
+// Temp LED
+unsigned long previousMillis = 0;       // will store last time LED was updated
```
Comment (kjyv) [inline src/rancilio-pid.cpp:207]: previousMillis is a bad global variable name, too generic

File: src/rancilio-pid.cpp [hunk 5]
```diff
@@ -1675,22 +1681,87 @@ void debugVerboseOutput() {
             bPID.GetKd());
     }
 }
+/**
+ * @brief blink LED to intervall
+ */
+void blinkLED(){
+    unsigned long currentMillis = millis();
+    if (currentMillis - previousMillis >= LEDInterval)
```
Comment (kjyv) [inline src/rancilio-pid.cpp:1689]: here and following, put { inline

File: src/rancilio-pid.cpp (hunk 2 above)
Comment (tml89) [reply src/rancilio-pid.cpp]: fixed

File: src/rancilio-pid.cpp [hunk 6]
```diff
@@ -1423,22 +1430,27 @@ void machinestatevoid() {
 
         case kBrew:
             brewdetection();
+            // Ausgabe waehrend des Bezugs von Bruehzeit, Temp und heatrateaverage
```
Comment (kjyv) [inline src/rancilio-pid.cpp:1433]: It seems you've reverted some code back to an old state (incl. german comment etc.)

**Comment by tml89** [comment]
due the changes in master. Close PR and resubmit

**Comment by murmeltier08** [comment]
Thank you !:)

**Pull request closed without merging**